renpy/loader.py

```diff
@@ -400,10 +439,13 @@ def scandirfiles_from_archives(add, seen):
     Scans archives and fills out game_files.
     """
 
-    files = game_files
-
     for _prefix, index in archives:
         for j in index:
+            # Check if the file should be treated as a common file.
+            if len(index[j]) > 3 and index[j][3] is True:
```

Comment:
I don't like the len() here. It might make sense to store the common vs not-common distinction in archives, and then do:

```
for _prefix, index, common in archives:
    if common:
         files = common_files 
    else:
         files = game_files 
```

(And the same thing down around line 722.)